### PR TITLE
Restore RRule support in amp-date-picker

### LINKED ISSUE #24682: amp-date-picker highlighted/blocked attributes do not work
## What's the issue?

amp-date-picker highlighted and blocked attributes have no effect

## How do we reproduce the issue?

See https://amp.dev/documentation/examples/components/amp-date-picker/?format=websites#highlighted-and-blocked-attributes . 

## What browsers are affected?

All browsers? 

## Which AMP version is affected?

AMP version 1909141409590. Not sure if this is a new issue, haven't tried to use the date picker before.

### PR DESCRIPTION
**Pull request by caroqliu**
Reverts #25230, fixes #26154, fixes #24682. This PR replaces #26154.

This change restores `RRule` @ `2.6.4` whereas it was removed at `2.6.2`

### CODE REVIEW

**Comment by samouri** [comment]
I have some questions on our `rrule` usage / this PR. 

I know that we've been having issues with it due to luxon's [circular dependencies](https://github.com/moment/luxon/issues/193). Are we actually taking advantage of timezone support? rrule [externs](https://github.com/jakubroztocil/rrule/blob/master/webpack.config.js#L57) the library and it seems like it is meant to be an optional dependency (and run fine without it).

If we aren't taking advantage of timezone support, shouldn't we also extern luxon? This would have the benefits of:
- fixing our circ deps without complicating the build
- ensuring we don't include `luxon`'s code if we don't use it (closure compiler _may_ have DCEd it anyway)

File: build-system/externs/dompurify.extern.js [hunk 1]
```diff
@@ -51,3 +51,9 @@ DomPurifyConfig.prototype.ALLOWED_URI_REGEXP;
 DomPurifyConfig.prototype.ADD_TAGS;
 DomPurifyConfig.prototype.ADD_ATTR;
 DomPurifyConfig.prototype.ADD_URI_SAFE_ATTR;
+
+/**
+ * @typedef {{addHook: !Function, removeAllHooks: !Function, sanitize: !Function}}
+ */
+var DomPurifyDef;
```
Comment (samouri) [inline build-system/externs/dompurify.extern.js]: Is this meant to be part of this PR?

**Comment by mrjoro** [comment]
Is this PR still active?

**Comment by caroqliu** [comment]
> Is this PR still active?

Sadly yes, haven't had an opportunity to revisit recently but I'll do a fresh merge and see where we are at. Thanks for keeping me honest, Joey. 😄

**Pull request closed without merging**